types/types.go

```diff
@@ -292,6 +292,8 @@ type SystemContext struct {
 	// Note that this field is used mainly to integrate containers/image into projectatomic/docker
 	// in order to not break any existing docker's integration tests.
 	DockerDisableV1Ping bool
+	// Directory to use for temporary files
+	TmpDirPath string
```

Comment:
For the public value, please name it `OSTreeTmpDirPath` or something like that; it has a specific OSTree-specific purpose (SELinux labels) and we do not use it in general.

(Having a global override of the `temporaryDirectoryForBigFiles` constant _could_ be useful, but the point of this one is the specific OSTree semantics, not a replacement for `$TMPDIR`.)